src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+  render(<Textarea id="foo" label="bar" fieldHelp="baz" />);
+
+  expect(screen.getByText("baz")).toHaveAttribute("id", "foo-field-help");
+  expect(screen.getByRole("textbox")).toHaveAttribute(
+    "aria-describedby",
+    "foo-field-help"
+  );
+});
+
+test("fieldHelp is present and id is not present, the id of the field help is added to 'aria-describedby' in the input", () => {
+  render(<Textarea label="bar" fieldHelp="baz" />);
+
+  expect(screen.getByText("baz")).toHaveAttribute(
+    "id",
+    `${mockedGuid}-field-help`
+  );
+  expect(screen.getByRole("textbox")).toHaveAttribute(
+    "aria-describedby",
+    `${mockedGuid}-field-help`
+  );
+});
+
+test.each(["info", "warning", "error"])(
+  "with id prop provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(
+      <Textarea
+        id="foo"
+        label="bar"
+        fieldHelp="baz"
+        {...{ [validationType]: "test" }}
+      />
+    );
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(screen.getByRole("tooltip")).toHaveAttribute("id", "foo-validation");
+    expect(textarea).toHaveAttribute(
+      "aria-describedby",
+      "foo-field-help foo-validation"
+    );
+  }
+);
+
+test.each(["info", "warning", "error"])(
+  "with id prop not provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(
+      <Textarea label="bar" fieldHelp="baz" {...{ [validationType]: "test" }} />
+    );
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(screen.getByRole("tooltip")).toHaveAttribute(
+      "id",
+      `${mockedGuid}-validation`
+    );
+    expect(textarea).toHaveAttribute(
+      "aria-describedby",
+      `${mockedGuid}-field-help ${mockedGuid}-validation`
+    );
+  }
+);
+
+describe(`when the characterLimit prop is passed`, () => {
+  it.each([2, 3, 4])("renders a character counter", (characterLimit) => {
+    const valueString = "foo";
+    const limitMinusValue = characterLimit - valueString.length >= 0;
+    render(<Textarea value={valueString} characterLimit={characterLimit} />);
+    const underCharacters =
+      characterLimit - valueString.length === 1 ? "character" : "characters";
+    const overCharacters =
+      valueString.length - characterLimit === 1 ? "character" : "characters";
+
+    expect(
+      screen.getByText(
+        `${
+          limitMinusValue
+            ? `${characterLimit - valueString.length} ${underCharacters} left`
+            : `${
+                valueString.length - characterLimit
+              } ${overCharacters} too many`
+        }`,
+        { selector: '[aria-hidden="true"]' }
+      )
+    ).toBeInTheDocument();
+  });
```

Comment:
**question**: is it necessary to test all three scenarios here? Splitting this test to handle the singular and plural use cases separately might simplify things.

Additionally, removing the conditional logic in this test could enhance readability. What do you think?